### DISCUSSION TITLE
Add configuration using the UI instead of / in addition to oldschool .yaml

### DISCUSSION THREAD

**Discussion by bcutter** [mampfes/hacs_waste_collection_schedule#565] [Ideas]
All said in the title - and saying (wishing) is easy, but the implementation for sure is quite some work.

Anyway, that would open this great integration to a much broader group of users.
- Currently it's just very/way to complicated to set this up with a ICS and sensors and stuff. And needing to restart whole HA on every change instead of being able to simply reload the integration from the ui is also very time-consuming and frustrating.
- The simple fact that handling with yaml is necessary will for sure lower the motivation of many people using this at all. So they never experience the power of this integration :-)

**Comment by mampfes** [comment]
I totally agree - a UI based configuration would be very nice. But there are so many configuration options already that I think this is simply not possible - at least if we want to have exactly the same functionality as today.

What I can imagine is an intermediate step: I think that the sensor configuration could be made optional if WCS creates some sensors by default:
I think of one sensor which is responsible for all waste types. The primary value could be either the date of the next collection or the days to the next collection. In the extended attributes the waste type could be added.
In addition, WCS would create one sensor per discovered waste type with the same value and attributes as above.

But I don't not have good answer for some issues:
- How to hide unwanted waste types? The waste type specific sensors could be disabled via the UI. But how to remove them from the all-waste-types-sensor. If it is possible the get the disabled state, this information could be used.
- How to deal with the alias functionality?
- How to deal with waste type specific icons (which can be overwritten using the existing sensor configuration)?
- The user has to use template sensor to sensors with specific text (or use the existing textual sensors configuration).
- How to deal with multiple sources?
- How to deal with the source specific argument?

Maybe the way to go is a to offer a subset of the functionality via the UI and the power-users have to use text based configuration. But where is the border line?

**Reply by szaske** [reply]
I agree. I'd like to get this working with Seattle Waste collection, which looks like it's supported.  I'm able to install the code but the instructions do not explain how to add it to a dashboard, or how to get the visual shown on the projects home page.  I feel like I'm a pretty advanced HA user and I cannot figure out.

**Comment by 5ila5** [comment]
You might want to take a look here: #2068
